SpyrosSiannas/record-label-webapp
# Db python backend

### Based on #38, wait for a PR from that branch before merging this. Requires changes for the DB content and creation sql

Added Python Backend
---
### Basic run-through
The way this works is via two core objects and some helper scripts
- The `Database` class, creates a connection and sets the database cursors (which line / table the sql points to). It's used to get the connection and cursors
- The `Controller` class, which is initalised with a database object and is used to run the queries 
- The `fromfile` script that is used to load the sql commands from the `sql_queries` directory by filename
- The cfg file that contains some global configuration parameters like the DB name and Password

### Cool ascii art for the script
![image](https://user-images.githubusercontent.com/59615151/120105016-e3a18780-c15f-11eb-8e1f-b06d36d5dbc9.png)

How to test!
---
- First take a look at the instructions inside the `database` readme.md directory (github should display it automatically if you navigate to there).
- After you go through the instructions, you should try running all of the functions 
**NOTE:** Empty the DB first and _**then**_ create it
